tests/test_data/test_datasets/test_builder.py

```diff
@@ -0,0 +1,57 @@
+import pytest
```

Comment:
Thank you for providing the  unit test code for build_dataset. There are still some problems when runing unit test and we highly suggest you to run `pytest` in your local machine before pushing your updated code to this PR.

Reply:
Yes, I am trying. The latest commit works on my local machine. Please check it.

Reply:
All checks have passed. Thank you ~